geniusyield-market-maker/test/GeniusYield/Test/MarketMaker/Utils.hs

```diff
@@ -0,0 +1,149 @@
+module GeniusYield.Test.MarketMaker.Utils where
+
+import           Data.List                         (isInfixOf, findIndex)
+import           Control.Concurrent.MVar
+import           Control.Exception                 (IOException, throwIO, catch, displayException)
+import           Control.Monad                     (foldM)
+import           Control.Monad.State               (StateT)
+import           GeniusYield.MarketMaker.Constants (logNS)
+import           GeniusYield.MarketMaker.Prices    (PricesProviderCfg(..), MockConfig)
+import           GeniusYield.Types
+
+
+data PPStatus = PPStatus { pps1Offset :: Double, pps1Available :: Bool, pps2Available :: Bool }
+  deriving stock Show
+
+normalPPStatus :: PPStatus
+normalPPStatus = PPStatus { pps1Offset = 0, pps1Available = True, pps2Available = True }
+
+newtype MBLog = MBLog (GYLogSeverity, GYLogNamespace, String)
+  deriving newtype Show
+
+data LogData = LDStatus PPStatus | LDLog MBLog
+  deriving stock Show
+
+augmentedLogRun :: MVar [LogData]
+                -> (GYLogNamespace -> GYLogSeverity -> String -> IO ())
+                -> GYLogNamespace -> GYLogSeverity -> String
+                -> IO ()
+augmentedLogRun logRef f ns s msg = do
+  modifyMVar_ logRef $ \current -> pure $ current ++ [LDLog (MBLog (s, ns, msg))]
```

Comment:
Prepending is faster (O(1)) for Haskell prelude list.

Reply:
Done.